lib/spack/spack/binary_distribution.py

```diff
@@ -127,7 +134,18 @@ def write_buildinfo_file(prefix, workdir, rel=False):
             #  Check if the file contains a string with the installroot.
             #  This cuts down on the number of files added to the list
             #  of files potentially needing relocation
-            if relocate.strings_contains_installroot(
+            if os.path.islink(path_name):
+                link = os.readlink(path_name)
+                if os.path.isabs(link):
+                    if link.startswith(spack.store.layout.root):
+                        rel_path_name = os.path.relpath(path_name, prefix)
+                        link_to_relocate.append(rel_path_name)
+                    else:
+                        msg = 'Absolute link %s to %s ' % (path_name, link)
```

Comment:
Would you want to use `UnrelocatableLinkException` here? It is not currently used anywhere.

Reply:
@tgamblin asked me to change this from an error to a warning; UnrelocatableLinkException was written to be used here, and I will remove it now that it no longer is used here.

Reply:
yeah warning is the way to go here for now.  If you rely on something in the OS intentionally (e.g. something in /opt?), and we mark the binary properly, this is still a legit case.  We may want to whitelist paths that shouldn't generate warnings.... I think we have to see how it's used first, though.